src/backend/distributed/commands/alter_table.c

```diff
@@ -1418,8 +1418,14 @@ CheckAlterDistributedTableConversionParameters(TableConversionState *con)
 
 		Var *colocateWithPartKey = DistPartitionKey(colocateWithTableOid);
 
-		if (con->distributionColumn &&
-			colocateWithPartKey->vartype != con->distributionKey->vartype)
+		if (colocateWithPartKey == NULL)
+		{
+			/* this should never happen */
+			ereport(ERROR, (errmsg("table to colocate with doesn't have a "
```

Comment:
might be good to print both table names here (input table & the table which we are trying to colocate with)

also, might worth to have a regression test